addons/mail/static/tests/messaging_menu/messaging_menu.test.js

```diff
@@ -725,6 +727,38 @@ test("channel preview: basic rendering", async () => {
     await contains(".o-mail-NotificationItem-text", { text: "Demo: test" });
 });
 
+test("chat preview should not display correspondent name in body", async () => {
+    // DM chat with demo, the conversation is named "Demo" and body is simply message content
+    // not prefix like "Demo:"
+    const pyEnv = await startServer();
+    const partnerId = pyEnv["res.partner"].create({ name: "Demo", email: "demo@odoo.com" });
+    const userId = pyEnv["res.users"].create({ partner_id: partnerId });
+    const channelId = pyEnv["discuss.channel"].create({
+        channel_type: "chat",
+        channel_member_ids: [
+            Command.create({ partner_id: serverState.partnerId }),
+            Command.create({ partner_id: partnerId }),
+        ],
+    });
+    await start();
+    await click(".o_menu_systray .dropdown-toggle:has(i[aria-label='Messages'])");
+    await withUser(userId, () =>
+        rpc("/mail/message/post", {
+            post_data: {
+                body: "<p>test</p>",
+                message_type: "comment",
+            },
+            thread_id: channelId,
+            thread_model: "discuss.channel",
+        })
+    );
```

Comment:
We should avoid using `withUser` as much as possible (user changing for the whole mock server, not only within the callback scope which can lead to inconsistencies) especially since a simple message create does the trick :)

Reply:
I'm aware of `withUser` issues... But I need to test with posting of message from someone else, and I didn't want to put `email_from` in data that was there to check it was not being taken into account. This is not okay to test negativity that way.

Here we have `email` field set on partner and we make the user genuinely post message, so checking we use the name (and not email, thanks to exact check) is more reliable

Reply:
It's more reliable if you assume the mock server is properly implemented, which is not always the case, let's face it. Posting a message to get the `email_from` key or creating a message with the `email_from` key is the same, except that we don't rely on a known source of non-determinism, and it is shorter.